applications/MeshingApplication/python_scripts/mmg_process.py

```diff
@@ -417,12 +419,38 @@ def ExecuteInitializeSolutionStep(self):
                                 self.step = 0  # Reset
                                 self.time = 0.0  # Reset
 
+    def ExecuteFinalize(self):
+        """ This method is executed in order to finalize the simulation and save the refined mesh in a new .mdpa file
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+        self.average_remeshing = self.settings["remesh_post_process"].GetBool()
+        output_file_path = self.settings["output_file_name"].GetString()
+        if self.average_remeshing:
+            self._ExecuteRefinement()
+            if self.main_model_part.HasSubModelPart("fluid_computational_model_part"):
+                self.main_model_part.RemoveSubModelPart("fluid_computational_model_part")
```

Comment:
This is needed to make sure that the model_part that we output is ready to be run again, and in some cases we need to remove some model parts.

However, this is too specific for the FluidDynamicsApplications so we need to write this in a more general way.

```suggestion
            for sub_model_part_name in self.sub_model_part_names_to_remove:
                if self.main_model_part.HasSubModelPart(sub_model_part_name):
                    self.main_model_part.RemoveSubModelPart(sub_model_part_name)
```

Where `self.sub_model_part_names_to_remove:` is written in the settings.

Reply:
This part has been modified in a more general one, adding the string in the default parameters